pkg/controller/servicediscovery/privatednsnamespace/setup.go

```diff
@@ -0,0 +1,55 @@
+package privatednsnamespace
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	svcsdk "github.com/aws/aws-sdk-go/service/servicediscovery"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/servicediscovery/v1alpha1"
+)
+
+// SetupPrivateDNSNamespace adds a controller that reconciles PrivateDNSNamespaces.
+func SetupPrivateDNSNamespace(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(svcapitypes.PrivateDNSNamespaceGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.observe = e.observeImpl
+			e.preCreate = preCreate
+			e.postCreate = nopPostCreate
+			e.delete = e.deleteImpl
+			e.update = e.updateImpl
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.PrivateDNSNamespace{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.PrivateDNSNamespaceGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+// ActualIsNotFound reimplements IsNotFound which doesn't do it's job
+func ActualIsNotFound(err error) bool {
```

Comment:
Those two functions don't belong to `setup.go` and should rather be in `hooks.go` where they are used.

Reply:
As far as I'm concerned they should be part of the generated code, which is why I've placed them in the setup.go.
Moved to hooks.go.